// react-options/lib/index.js


exports.__esModule = true;

const _react = require('react');

const _react2 = _interopRequireDefault(_react);

const _propTypes = require('prop-types');

const _propTypes2 = _interopRequireDefault(_propTypes);

const _classnames = require('classnames');

const _classnames2 = _interopRequireDefault(_classnames);

require('./style.css');

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError('Cannot call a class as a function');
  }
}

function _possibleConstructorReturn(self, call) {
  if (!self) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }
  return call && (typeof call === 'object' || typeof call === 'function') ? call : self;
}

function _inherits(subClass, superClass) {
  if (typeof superClass !== 'function' && superClass !== null) {
    throw new TypeError(
      `Super expression must either be null or a function, not ${typeof superClass}`
    );
  }
  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: { value: subClass, enumerable: false, writable: true, configurable: true }
  });
  if (superClass) {
    Object.setPrototypeOf
      ? Object.setPrototypeOf(subClass, superClass)
      : (subClass.__proto__ = superClass);
  }
}

const ReactOptions = (function (_PureComponent) {
  _inherits(ReactOptions, _PureComponent);

  function ReactOptions() {
    _classCallCheck(this, ReactOptions);

    return _possibleConstructorReturn(this, _PureComponent.apply(this, arguments));
  }

  ReactOptions.prototype.componentWillReceiveProps = function componentWillReceiveProps() {};

  ReactOptions.prototype.render = function render() {
    let _props = this.props,
      children = _props.children,
      details = _props.details,
      optionsActive = _props.optionsActive,
      id = _props.id;

    const detailsStyle = (0, _classnames2.default)('ro-details', {
      'ro-details_show': optionsActive === id
    });

    return _react2.default.createElement(
      'div',
      { className: 'ro-wrapper' },
      _react2.default.createElement('div', { className: 'ro-main' }, children),
      _react2.default.createElement('div', { className: detailsStyle }, details)
    );
  };

  return ReactOptions;
}(_react.PureComponent));

exports.default = ReactOptions;

ReactOptions.propTypes =
  process.env.NODE_ENV !== 'production'
    ? {
      children: _propTypes2.default.node.isRequired,
      details: _propTypes2.default.node.isRequired,
      optionsActive: _propTypes2.default.string,
      id: _propTypes2.default.string.isRequired
    }
    : {};

ReactOptions.defaultProps = {
  optionsActive: null
};
module.exports = ReactOptions;
